azurerm/resource_arm_public_ip.go

```diff
@@ -84,6 +84,15 @@ func resourceArmPublicIp() *schema.Resource {
 				Computed: true,
 			},
 
+			"sku": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Default:          string(network.PublicIPAddressSkuNameBasic),
+				ForceNew:         true,
+				ValidateFunc:     validatePublicIpSku,
```

Comment:
This could just be `StringInSlice` couldn't it?

Reply:
@paultyng I was mimicking the other validation functions in that file and figured it could be reused in the acceptance tests, similar to:
- https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_public_ip_test.go#L41
- https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_public_ip_test.go#L73

Is there a preferred approach?

Reply:
It depends what's being tested - but in general we're using the Constants/Enums in the SDK via the `StringInSlice` (at least for Strings) for newer resources; given these constants exist - could we switch this for a `StringInSlice`?